CERT-2991 (#126)
Certora/Documentation: docs/gambit/mutation-verifier.md

dependencies, run Gambit from the command line:

```
certoraMutate --prover_conf path/to/prover/prover.conf --mutation_conf path/to/mutation/mutation.conf
```

```{note}
You must run `certoraMutate` from the root of the Solidity project directory.
The files `prover.conf` and `mutation.conf`

> can be in their own directories.

Reviewer: There is an unnecessary line break before this

Author: not in the docs though?

All paths in `mutation.conf` are relative to the parent directory containing `mutation.conf`.

> This is different from how `prover.conf` is written, where the paths are all relative to the root

Reviewer: ```suggestion
Paths in `prover.conf` are all relative to the project directory's root, which is assumed to be the working directory.
```

}
```

### Manual Mutations
You can add manual mutations to `mutation.conf` like so:

```json
{
  "filename" : "C.sol",
  "num-mutants": 5,
  "manual_mutants": {
     "C.sol": "path/to/dir/with/manual_mutants/for/C"
  }
}
```

> If you set `num_mutants` to 0 in the above file, `gambit` will not generate any mutants, you will only run

Reviewer: I think `Gambit` should be capitalized as it is a name? Is `gambit` the command that is run?

Author: yes it's `gambit` not `Gambit`

Author: i think we should use `gambit` because that is what we are running, no?

> If you set `num_mutants` to 0 in the above file, `gambit` will not generate any mutants, you will only run

Reviewer: ```suggestion
If you set `num_mutants` to 0 in the above file, `gambit` will run only on the mutants added manually and no other mutants will be generated.
```

| Option                         | Description                                                                                                           |
|:-------------------------------|:----------------------------------------------------------------------------------------------------------------------|
| `--prover_conf`                | specify the Prover configuration file for verifying mutants                                                           |
| `--mutation_conf`              | specify the configuration file for mutant generation                                                                  |
| `--num_mutants`                | request the mutant generator to generate a specific number of mutants. Defaults to 5                                  |
| `--prover_version`             | specify the version of `certoraRun` to use for verification. Defaults to the installed version of `certoraRun`        |

> | `--server`                     | specify the server environment to run on. Defaults to the value specified in the file of `--prover_conf`, if the field exists otherwise whatever `certoraRun` uses by default   |

Reviewer: ```suggestion
| `--server`                     | specify the server environment to run on. Defaults to the value specified in the file of `--prover_conf`, or the `certoraRun` default otherwise.   |
```

| `--debug`                      | show additional logging information during execution                                                                  |
| `--gambit_out`                 | specify the output directory for gambit. Defaults to a new directory which is added in the working directory          |
| `--applied_mutants_dir`        | specify the target directory for mutant verification build files. Defaults to a hidden Prover internal directory      |
| `--ui_out`                     | specify a JSON file to dump the mutant verification report used for the web UI                                   |
| `--dump_link`                  | specify a text file to write the UI report link                                                                       |

> | `--dump_csv`                   | specify a csv file to write the verification JSON report                                                              |

Reviewer: If it is a csv file, it is not a JSON file - something is contradictory here

Author: this was a weird sentence. I meant to say that the content is similar to what is in the JSON. Fixed now.
